src/rhsmlib/facts/pkg_arches.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function, division, absolute_import
+
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+import sys
+import os
+import logging
+import platform
+
+from rhsmlib.facts import collector
+
+log = logging.getLogger(__name__)
+
+
+class SupportedArchesCollector(collector.FactsCollector):
+    """
+    Class used for collecting packages architectures of a host
+    """
+
+    def __init__(self, arch=None, prefix=None, testing=None):
+        super(SupportedArchesCollector, self).__init__(
+            arch=arch,
+            prefix=prefix,
+            testing=testing)
+
+
+    def get_arches_on_debuntu(self):
+        """
+        Try to return content of all supported packages architectures
+        :return: dictionary containing architectures
+        Otherwise empty dictionary is returned.
+        """
+        arches = []
+        ARCH_FILE_NAME = '/var/lib/dpkg/arch'
+        arch_file = os.path.join(self.prefix, ARCH_FILE_NAME)
+        try:
+            with open(arch_file) as f:
+                 arches = [line.rstrip() for line in f]
+        except IOError as e:
+            # If we specify a prefix, and there is no 'arch' file,
+            # consider that fatal.
+            log.exception(e)
+            raise
+        return {
+            'supported_architectures': arches
+        }
+
+    def get_all(self):
+        dist_name, dist_vers, _codename = platform.linux_distribution(full_distribution_name=False)
+        if dist_name == 'Ubuntu' or dist_name == 'Debian':
+            return self.get_arches_on_debuntu()
+
+
+if __name__ == '__main__':
+    _LIBPATH = "/usr/share/rhsm"
+    from pprint import pprint
+
+    # add to the path if need be
+    if _LIBPATH not in sys.path:
+        sys.path.append(_LIBPATH)
+
+    from subscription_manager import logutil
```

Comment:
This module `logutil` was moved from `subscription_manager` package to `rhsm` package.